src/subdomains/core/referral/reward/ref-reward-out.service.ts

```diff
@@ -0,0 +1,123 @@
+import { Injectable } from '@nestjs/common';
+import { PayoutOrderContext } from 'src/subdomains/supporting/payout/entities/payout-order.entity';
+import { PayoutRequest } from 'src/subdomains/supporting/payout/interfaces';
+import { PayoutService } from 'src/subdomains/supporting/payout/services/payout.service';
+import { RefRewardRepository } from './ref-reward.repository';
+import { RefReward, RewardStatus } from './ref-reward.entity';
+import { AssetService } from 'src/shared/models/asset/asset.service';
+import { UserService } from 'src/subdomains/generic/user/models/user/user.service';
+import { LiquidityOrderContext } from 'src/subdomains/supporting/dex/entities/liquidity-order.entity';
+import { DexService } from 'src/subdomains/supporting/dex/services/dex.service';
+
+@Injectable()
+export class RefRewardOutService {
+  constructor(
+    private readonly refRewardRepo: RefRewardRepository,
+    private readonly payoutService: PayoutService,
+    private readonly assetService: AssetService,
+    private readonly userService: UserService,
+    private readonly dexService: DexService,
+  ) {}
+
+  async payoutTransactions(): Promise<void> {
+    try {
+      const transactionsPaidOut = await this.refRewardRepo.find({
+        where: { status: RewardStatus.PAYING_OUT },
+        relations: ['user'],
+      });
+
+      await this.checkCompletion(transactionsPaidOut);
+
+      const transactionsToPayout = await this.refRewardRepo.find({
+        where: { status: RewardStatus.READY_FOR_PAYOUT },
+        relations: ['user'],
+      });
+
+      // pay out ref rewards
+      const successfulRequests = [];
+
+      for (const transaction of transactionsToPayout) {
+        try {
+          await this.doPayout(transaction);
+          successfulRequests.push(transaction);
+
+          await this.updatePaidRefCredit([transaction.user?.id]);
+        } catch (e) {
+          console.error(`Failed to initiate buy-crypto payout. Transaction ID: ${transaction.id}`);
+          // continue with next transaction in case payout initiation failed
+          continue;
+        }
+      }
+
+      this.logTransactionsPayouts(successfulRequests);
+    } catch (e) {
+      console.error(e);
+    }
+  }
```

Comment:
I think I would split this method into two methods (e.g. payoutNewTransactions and checkPendingTransactions) which are called independently from ref-reward.service, makes it more readable.